src/python/pants/option/parser.py

```diff
@@ -316,15 +283,19 @@ def error(
         default_value = kwargs.get("default")
         if default_value is not None:
             if isinstance(default_value, str) and type_arg != str:
-                # attempt to parse default value, for correctness..
+                # attempt to parse default value, for correctness.
                 # custom function types may implement their own validation
                 default_value = self.to_value_type(default_value, type_arg, member_type)
                 if hasattr(default_value, "val"):
                     default_value = default_value.val
 
                 # fall through to type check, to verify that custom types returned a value of correct type
 
-            if isinstance(type_arg, type) and not isinstance(default_value, type_arg):
+            if (
+                isinstance(type_arg, type)
+                and not isinstance(default_value, type_arg)
+                and not (issubclass(type_arg, bool) and default_value == UnsetBool)
```

Comment:
Previously validation tripped up on UnsetBool. I think the only reason this didn't blow up is that we don't actually use UnsetBool anywhere, and the tests were not validating. But now that they are, they caught this.

Possibly we can get rid of UnsetBool, I don't see why None couldn't be used, but one thing at a time.